modules/authenticate/src/actions/sign-in.ts

```diff
@@ -227,14 +225,67 @@ export const getAuthenticatedUser = (idToken: string): AuthenticatedUserInterfac
     };
 };
 
+/**
+ * Send account switch request.
+ *
+ * @param {AccountSwitchRequestParams} requestParams request parameters required for the account switch request.
+ * @param {string} clientHost client host.
+ * @returns {Promise<TokenResponseInterface>} token response data or error.
+ */
+export const sendAccountSwitchRequest = (
+    requestParams: AccountSwitchRequestParams,
+    clientHost: string
+): Promise<TokenResponseInterface> => {
+    const tokenEndpoint = getTokenEndpoint();
+    if (!tokenEndpoint || tokenEndpoint.trim().length === 0) {
+        return Promise.reject(new Error("Invalid token endpoint found."));
+    }
+
+    const body = [];
+    body.push(`grant_type=account_switch`);
+    body.push(`username=${ requestParams.username }`);
+    body.push(`userstore-domain=${ requestParams["userstore-domain"] }`);
+    body.push(`tenant-domain=${ requestParams["tenant-domain"] }`);
+    body.push(`token=${ getSessionParameter(ACCESS_TOKEN) }`);
+    body.push(`scope=openid`);
+    body.push(`client_id=${ requestParams.client_id }`);
+
+    return axios.post(tokenEndpoint, body.join("&"), getTokenRequestHeaders(clientHost))
+        .then((response) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Invalid status code received in the token response: "
+                    + response.status));
+            }
```

Comment:
Is it necessary to check for the status here?

Reply:
I think it would be better to check the status here to keep the requests consistent with the swagger spec.